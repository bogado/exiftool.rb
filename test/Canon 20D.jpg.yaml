--- 
:color_temp_custom2: 4792
:sensor_left_border: 84
:flash_activity: 0
:measured_rggb: 674 1024 1024 670
:photo_effect: "Off"
:digital_zoom: None
:af_points_in_focus: 4
:color_temp_shade: 7000
:y_cb_cr_sub_sampling: YCbCr4:2:2 (2 1)
:metering_mode: Evaluative
:image_size: 3504x2336
:canon_model_id: EOS 20D
:bracket_mode: "Off"
:camera_type: EOS Mid-range
:exposure_mode: Auto
:tone_curve: Standard
:af_area_x_positions: 8 -555 571 -931 8 947 -555 571 8
:color_tone: Normal
:af_image_width: 3504
:max_aperture: 5.6
:aeb_bracket_value: 0
:shutter_curtain_sync: 1st-curtain sync
:aeb_sequence_auto_cancel: 0,-,+/Enabled
:scale_factor35efl: 1.6
:lens_af_stop_button: AF stop
:add_original_decision_data: "Off"
:thumbnail_image: (Binary data 11259 bytes)
:flash_sync_speed_av: Auto
:black_mask_right_border: 0
:base_iso: 200
:image_width: 3504
:focal_plane_x_resolution: 2886.363636
:exposure_program: Program AE
:safety_shift_in_av_or_tv: Disable
:measured_ev: 13.25
:continuous_drive: Single
:red_balance: 1.943842
:file_name: Canon 20D.jpg
:white_balance_blue: 0
:lens: 17.0 - 85.0 mm
:digital_gain: 0
:macro_mode: Normal
:wb_rggb_levels_flash: 2187 1023 1023 1333
:menu_button_display_position: Previous (top if power off)
:sensor_height: 2360
:file_size: 2.9 MB
:black_mask_top_border: 0
:ettlii: Evaluative
:af_area_width: 78
:f_number: 9.0
:directory: test
:black_mask_bottom_border: 0
:focal_plane_y_size: 15.37 mm
:toning_effect: None
:af_area_y_positions: 504 270 270 4 4 4 -262 -262 -496
:focal_plane_y_resolution: 2885.805763
:target_aperture: 9
:focus_mode: One-shot AF
:mirror_lockup: Disable
:color_temp_flash: 6309
:make: Canon
:set_function_when_shooting: Image replay
:wb_bracket_value_gm: 0
:exposure_time: !ruby/object:Rational 
  denominator: 250
  numerator: 1
:thumbnail_offset: 9728
:canon_flash_mode: "Off"
:serial_number: 0330113864
:fov: 15.4 deg
:thumbnail_image_valid_area: 0 159 7 112
:af_point_selection_method: Normal
:color_space: sRGB
:original_decision_data_offset: 0
:lens_id: Unknown 17-85mm
:shutter_speed: !ruby/object:Rational 
  denominator: 250
  numerator: 1
:light_value: 13.3
:superimposed_display: "On"
:exposure_compensation: 0
:orientation: Horizontal (normal)
:components_configuration: Y, Cb, Cr, -
:"shutter-ae_lock": AF/AE lock
:focal_length: 85.0 mm
:focal_length35efl: "85.0 mm (35 mm equivalent: 132.8 mm)"
:bracket_shot_number: 0
:model: Canon EOS 20D
:aperture_value: 9.0
:flash_guide_number: 0
:long_exposure_noise_reduction2: "Off"
:focal_plane_x_size: 23.04 mm
:sequence_number: 0
:long_exposure_noise_reduction: "Off"
:white_balance: Auto
:file_type: JPEG
:canon_exposure_mode: Program AE
:interop_version: "0100"
:canon_image_width: 3504
:color_temp_fluorescent: 3952
:canon_firmware_version: Firmware 1.0.2
:sharpness: "+1"
:color_temp_auto: 4662
:af_assist_beam: Emits
:iso: 200
:black_mask_left_border: 0
:color_temp_custom1: 6163
:focal_plane_resolution_unit: inches
:wb_rggb_levels_custom1: 2095 1023 1023 1316
:flash_bits: (none)
:sensor_right_border: 3587
:bracket_value: 0
:canon_image_type: Canon EOS 20D
:exif_version: "0221"
:exif_image_width: 3504
:encoding_process: Baseline DCT, Huffman coding
:contrast: "+1"
:file_modify_date: 2012-02-19 21:49:48 -08:00
:white_balance_red: 0
:auto_rotate: None
:scene_capture_type: Standard
:compression: JPEG (old-style)
:sensor_red_level: 0
:control_mode: Camera Local Control
:sensor_width: 3596
:shooting_mode: Program AE
:lens_type: Unknown (-1)
:sensor_blue_level: 0
:file_permissions: rw-r--r--
:num_af_points: 9
:focus_continuous: Single
:color_temp_tungsten: 3196
:thumbnail_length: 11259
:target_exposure_time: !ruby/object:Rational 
  denominator: 251
  numerator: 1
:valid_af_points: 9
:long_focal: 85 mm
:wb_rggb_levels_daylight: 1995 1023 1023 1509
:interop_index: R98 - DCF basic file (sRGB)
:canon_image_height: 2336
:create_date: 2004-09-19 12:25:20 -07:00
:wb_bracket_mode: "Off"
:focal_units: 1/mm
:serial_number_format: Format 2
:short_focal: 17 mm
:sensor_top_border: 19
:measured_ev2: 13.5
:image_height: 2336
:color_temp_as_shot: 4662
:flash: Off, Did not fire
:iso_expansion: "On"
:record_mode: JPEG
:flashpix_version: "0100"
:wb_rggb_levels_tungsten: 1410 1023 1023 2408
:blue_balance: 1.638424
:sensor_bottom_border: 2354
:circle_of_confusion: 0.019 mm
:exif_byte_order: Little-endian (Intel, II)
:custom_rendered: Normal
:drive_mode: Single-frame Shooting
:color_components: 3
:x_resolution: 72
:exif_image_height: 2336
:aperture: 9.0
:easy_mode: Manual
:nd_filter: n/a
:exposure_level_increments: !ruby/object:Rational 
  denominator: 3
  numerator: 1
:auto_exposure_bracketing: "Off"
:af_area_height: 78
:flash_firing: Fires
:wb_rggb_levels_custom2: 1912 1023 1023 1627
:self_timer2: 0
:lens35efl: "17.0 - 85.0 mm (35 mm equivalent: 26.6 - 132.8 mm)"
:mime_type: image/jpeg
:wb_rggb_levels_shade: 2292 1023 1023 1267
:canon_image_size: Large
:hyperfocal_distance: 41.74 m
:wb_rggb_levels_auto: 1973 1023 1023 1663
:zoom_source_width: 0
:auto_iso: 100
:vrd_offset: 0
:wb_shift_gm: 0
:wb_rggb_levels_fluorescent: 1797 1023 1023 2058
:color_temperature: 4800
:wb_rggb_levels: 1973 1015 1015 1663
:wb_shift_ab: 0
:user_comment: ""
:zoom_target_width: 0
:quality: Fine
:color_temp_cloudy: 6000
:af_image_height: 2336
:manual_flash_output: n/a
:flash_exposure_comp: 0
:wb_rggb_levels_cloudy: 2160 1023 1023 1373
:source_file: test/Canon 20D.jpg
:color_temp_daylight: 5200
:sharpness_frequency: n/a
:slow_shutter: None
:modify_date: 2004-09-19 12:25:20 -07:00
:bulb_duration: 0
:bits_per_sample: 8
:focus_range: Not Known
:saturation: "+1"
:focal_type: Zoom
:wb_rggb_levels_as_shot: 1973 1015 1015 1663
:min_aperture: 32
:exif_tool_version: 8.59
:y_resolution: 72
:date_time_original: 2004-09-19 12:25:20 -07:00
:shutter_speed_value: !ruby/object:Rational 
  denominator: 250
  numerator: 1
:filter_effect: None
:resolution_unit: inches
:y_cb_cr_positioning: Co-sited
:picture_style: None
:file_number: 793-9390
:self_timer: "Off"
:wb_bracket_value_ab: 0
:optical_zoom_code: n/a
:owner_name: unknown
