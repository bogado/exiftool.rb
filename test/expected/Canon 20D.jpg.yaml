---
:file_name: Canon 20D.jpg
:file_size: 2.9 MB
:file_type: JPEG
:mime_type: image/jpeg
:exif_byte_order: Little-endian (Intel, II)
:make: Canon
:model: Canon EOS 20D
:orientation: Horizontal (normal)
:x_resolution: 72
:y_resolution: 72
:resolution_unit: inches
:y_cb_cr_positioning: Co-sited
:exposure_time: !ruby/object:Rational
  denominator: 250
  numerator: 1
:f_number: 9.0
:exposure_program: Program AE
:iso: 200
:exif_version: '0221'
:components_configuration: Y, Cb, Cr, -
:shutter_speed_value: !ruby/object:Rational
  denominator: 250
  numerator: 1
:aperture_value: 9.0
:flash: Off, Did not fire
:focal_length: 85.0 mm
:macro_mode: Normal
:self_timer: 'Off'
:quality: Fine
:canon_flash_mode: 'Off'
:continuous_drive: Single
:focus_mode: One-shot AF
:record_mode: JPEG
:canon_image_size: Large
:easy_mode: Manual
:digital_zoom: None
:contrast: '+1'
:saturation: '+1'
:sharpness: '+1'
:metering_mode: Evaluative
:focus_range: Not Known
:canon_exposure_mode: Program AE
:lens_type: Unknown (-1)
:max_focal_length: 85 mm
:min_focal_length: 17 mm
:focal_units: 1/mm
:max_aperture: 5.6
:min_aperture: 32
:flash_activity: 0
:flash_bits: (none)
:focus_continuous: Single
:zoom_source_width: 0
:zoom_target_width: 0
:photo_effect: 'Off'
:manual_flash_output: n/a
:focal_type: Zoom
:focal_plane_x_size: 23.04 mm
:focal_plane_y_size: 15.37 mm
:auto_iso: 100
:base_iso: 200
:measured_ev: 13.25
:target_aperture: 9
:target_exposure_time: !ruby/object:Rational
  denominator: 251
  numerator: 1
:exposure_compensation: 0
:white_balance: Auto
:slow_shutter: None
:sequence_number: 0
:optical_zoom_code: n/a
:flash_guide_number: 0
:flash_exposure_comp: 0
:auto_exposure_bracketing: 'Off'
:aeb_bracket_value: 0
:control_mode: Camera Local Control
:measured_ev2: 13.5
:bulb_duration: 0
:camera_type: EOS Mid-range
:auto_rotate: None
:self_timer2: 0
:canon_image_type: Canon EOS 20D
:canon_firmware_version: Firmware 1.0.2
:owner_name: unknown
:serial_number: 0330113864
:set_function_when_shooting: Image replay
:long_exposure_noise_reduction: 'Off'
:flash_sync_speed_av: Auto
:shutter-ae_lock: AF/AE lock
:af_assist_beam: Emits
:exposure_level_increments: 1/3 Stop
:flash_firing: Fires
:iso_expansion: 'On'
:aeb_sequence_auto_cancel: 0,-,+/Enabled
:superimposed_display: 'On'
:menu_button_display_position: Previous (top if power off)
:mirror_lockup: Disable
:af_point_selection_method: Normal
:ettlii: Evaluative
:shutter_curtain_sync: 1st-curtain sync
:safety_shift_in_av_or_tv: Disable
:lens_af_stop_button: AF stop
:add_original_decision_data: 'Off'
:canon_model_id: EOS 20D
:num_af_points: 9
:valid_af_points: 9
:canon_image_width: 3504
:canon_image_height: 2336
:af_image_width: 3504
:af_image_height: 2336
:af_area_width: 78
:af_area_height: 78
:af_area_x_positions: 8 -555 571 -931 8 947 -555 571 8
:af_area_y_positions: 504 270 270 4 4 4 -262 -262 -496
:af_points_in_focus: 4
:thumbnail_image_valid_area: 0 159 7 112
:serial_number_format: Format 2
:original_decision_data_offset: 0
:file_number: 793-9390
:bracket_mode: 'Off'
:bracket_value: 0
:bracket_shot_number: 0
:long_exposure_noise_reduction2: 'Off'
:wb_bracket_mode: 'Off'
:wb_bracket_value_ab: 0
:wb_bracket_value_gm: 0
:filter_effect: None
:toning_effect: None
:tone_curve: Standard
:sharpness_frequency: n/a
:sensor_red_level: 0
:sensor_blue_level: 0
:white_balance_red: 0
:white_balance_blue: 0
:color_temperature: 4800
:picture_style: None
:digital_gain: 0
:wb_shift_ab: 0
:wb_shift_gm: 0
:measured_rggb: 674 1024 1024 670
:vrd_offset: 0
:sensor_width: 3596
:sensor_height: 2360
:sensor_left_border: 84
:sensor_top_border: 19
:sensor_right_border: 3587
:sensor_bottom_border: 2354
:black_mask_left_border: 0
:black_mask_top_border: 0
:black_mask_right_border: 0
:black_mask_bottom_border: 0
:wb_rggb_levels_as_shot: 1973 1015 1015 1663
:color_temp_as_shot: 4662
:wb_rggb_levels_auto: 1973 1023 1023 1663
:color_temp_auto: 4662
:wb_rggb_levels_daylight: 1995 1023 1023 1509
:color_temp_daylight: 5200
:wb_rggb_levels_shade: 2292 1023 1023 1267
:color_temp_shade: 7000
:wb_rggb_levels_cloudy: 2160 1023 1023 1373
:color_temp_cloudy: 6000
:wb_rggb_levels_tungsten: 1410 1023 1023 2408
:color_temp_tungsten: 3196
:wb_rggb_levels_fluorescent: 1797 1023 1023 2058
:color_temp_fluorescent: 3952
:wb_rggb_levels_flash: 2187 1023 1023 1333
:color_temp_flash: 6309
:wb_rggb_levels_custom1: 2095 1023 1023 1316
:color_temp_custom1: 6163
:wb_rggb_levels_custom2: 1912 1023 1023 1627
:color_temp_custom2: 4792
:color_tone: Normal
:user_comment: ''
:flashpix_version: '0100'
:color_space: sRGB
:exif_image_width: 3504
:exif_image_height: 2336
:interop_index: R98 - DCF basic file (sRGB)
:interop_version: '0100'
:focal_plane_x_resolution: 2886.363636
:focal_plane_y_resolution: 2885.805763
:focal_plane_resolution_unit: inches
:custom_rendered: Normal
:exposure_mode: Auto
:scene_capture_type: Standard
:compression: JPEG (old-style)
:thumbnail_offset: 9728
:thumbnail_length: 11259
:image_width: 3504
:image_height: 2336
:encoding_process: Baseline DCT, Huffman coding
:bits_per_sample: 8
:color_components: 3
:y_cb_cr_sub_sampling: YCbCr4:2:2 (2 1)
:aperture: 9.0
:drive_mode: Single-frame Shooting
:image_size: 3504x2336
:lens: 17.0 - 85.0 mm
:lens_id: Unknown 17-85mm
:shooting_mode: Program AE
:shutter_speed: !ruby/object:Rational
  denominator: 250
  numerator: 1
:thumbnail_image: (Binary data 11259 bytes)
:wb_rggb_levels: 1973 1015 1015 1663
:blue_balance: 1.638424
:light_value: 13.3
:red_balance: 1.943842
